### PR TITLE
Deprecate NSOpenPanel and NSSavePanel constructors

### LINKED ISSUE #6474: NSOpenPanel and NSSavePanel no longer work on macOS 10.15
# Steps to Reproduce

Call the NSOpenPanel or NSSavePanel constructor. This only happens on macOS 10.15. 

### Expected Behavior

It will create the NSOpenPanel/NSSavePanel object.

### Actual Behavior

On macOS 10.15, it will fail with 'init returned null'.

### Environment

macOS 10.15
Xamarin.Mac
Calling mono from the command line. This also happens in my program which is built into a code-signed .app. It worked perfectly on all earlier versions of macOS.

### PR DESCRIPTION
**Pull request by wjk**
Does what the title says. Fixes #6474.

### CODE REVIEW

**Comment by spouliot** [comment]
build

**Comment by spouliot** [comment]
mono crashed while building
```
19:23:19 make[11]: *** [../../../class/lib/xammac_net_4_5/Facades/System.Drawing.Common.dll] Error 134
```
filed as https://github.com/xamarin/maccore/issues/1833
it's unrelated to the PR

**Comment by spouliot** [comment]
build

**Comment by rolfbjarne** [comment]
Test failure is unrelated

* monotouch-test: https://github.com/xamarin/maccore/issues/1834

**Comment by rolfbjarne** [comment]
Test failure is unrelated

* monotouch-test: https://github.com/xamarin/maccore/issues/1834

**Comment by rolfbjarne** [comment]
Great work, thanks!

File: src/appkit.cs [hunk 1]
```diff
@@ -8690,6 +8690,7 @@ interface NSOpenPanel {
 		NSOpenPanel OpenPanel { get; }
 
 		[Advice ("You must use 'OpenPanel' method if the application is sandboxed.")]
+		[Deprecated (PlatformName.MacOSX, 10, 15, message: "All open panels now run out-of-process, use 'OpenPanel' method instead")]
```
Comment (Therzok) [inline src/appkit.cs]: Can we also add `EditorBrowsable(EditorBrowsableState.Advanced)` or Never?

**Pull request merged**